### PR TITLE
Add build.zig.zon and update build.zig for the latest zig master 

### PR DESCRIPTION
**Pull request by dwclake**
## Changes:
- Adds a basic build.zig.zon file for use with zig's package manager.
- Updates build.zig for zig master 0.12.0-dev.3124+9e402704e, **potentially breaks compatibility with older versions of the zig compiler**.
- Creates llvm-zig as a static library linked with necessary libraries.
## How to use as a library:
- The following command will automatically add this module to an existing project's build.zig.zon file in the cwd, then the module just needs to be added in the projects build.zig file. Then llvm-zig will be built with the project.
```bash
$ cd /path/to/project
$ zig fetch --save https://github.com/kassane/llvm-zig/archive/main.tar.gz
```
```zig
// build.zig

// Create the dependency
const llvm = b.dependency("llvm-zig", .{});

// To add it to a library:
lib.root_module.addImport("llvm", llvm.module("llvm"));
// To add it to an executable:
exe.root_module.addImport("llvm", llvm.module("llvm"));
```
```zig
// main.zig
const llvm = @import("llvm");
```

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -1,60 +1,70 @@
...
-            .filepath = "examples/fatorial_module.zig",
-            .target = target,
+        buildExample(b, target, .{
+            .filepath = "examples/factorial_module.zig",
+            .target = target.query,
             .optimize = optimize,
         });
     }
+
+    const tests = b.option(bool, "Tests", "Build all tests [default: false]") orelse false;
```
Comment (kassane) [inline build.zig]: Why add `-DTests`?
There's already a b.step called test.
https://github.com/kassane/llvm-zig/blob/54c645b6bc87c557430a01aad7e269c8efcf8ec8/build.zig#L126

**Current:** `zig build test`
**Changed:** `zig build -DTests test`

Comment (dwclake) [reply build.zig]: It looked to me like the tests are being built even if you don't run the tests, so -DTests was added that the tests would not be built when using llvm-zig through the package manager. But I forgot about the workflow, which needs the test step being built so I'll revert it anyways.

**Comment by dwclake** [comment]
Changes:
- Rename llvm.zig to llvm-zig.zig
    - When zig was compiling llvm.zig in the test step it would error: llvm instrinsics cannot be defined, which appeared to be name clashes with the zig compiler's own llvm functions, renaming this file fixed failing tests.
- Remove -DTests

**Comment by kassane** [comment]
Hi @dwclake,

Amazing!
Thanks for your contribution to the project. Any improvements will always be welcome.

**Review by kassane — APPROVED** [review]

**Pull request merged**